client/src/features/ProjectPageV2/utils/AccessGuard.tsx

```diff
@@ -13,17 +13,27 @@
  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
- * limitations under the License
+ * limitations under the License.
  */
-import { Suspense, lazy } from "react";
-import PageLoader from "../../components/PageLoader";
 
-const ProjectV2Show = lazy(() => import("./show/ProjectV2Show"));
+import type { Role } from "../../projectsV2/api/projectV2.api.ts";
 
-export default function LazyProjectV2Show() {
-  return (
-    <Suspense fallback={<PageLoader />}>
-      <ProjectV2Show />
-    </Suspense>
-  );
+import { roleCompare } from "./roleUtils.ts";
+
+type AccessGuardProps = {
+  disabled: React.ReactNode;
+  enabled: React.ReactNode;
+  minimumRole?: Role;
+  role: Role;
+};
+
+export default function AccessGuard({
+  disabled,
+  enabled,
+  minimumRole = "owner",
+  role,
+}: AccessGuardProps) {
+  if (role === "owner") return enabled;
+  if (roleCompare(role, minimumRole, true) < 0) return disabled;
```

Comment:
This is unnecessarily hard to understand.

```suggestion
  if (toNumericRole(role) < toNumericRole(minimumRole)) return disabled;
```

Reply:
Addressed in cbda6a744409a0d51b978900e319cf0151162ed1